src/mscorlib/shared/System/String.cs

```diff
@@ -164,22 +446,23 @@ unsafe public char[] ToCharArray(int startIndex, int length)
             if (length > 0)
             {
                 char[] chars = new char[length];
-                fixed (char* src = &_firstChar) fixed (char* dest = chars)
+                fixed (char* src = &_firstChar)
+                fixed (char* dest = &chars[0])
                 {
                     wstrcpy(dest, src + startIndex, length);
                 }
                 return chars;
             }
-
             return Array.Empty<char>();
         }
 
-        public static bool IsNullOrEmpty(String value)
+        [NonVersionable]
```

Comment:
why this specifically needs such a marker?

Reply:
This attribute says that the implementation of this method is set in stone and that it is ok to inline it across modules for R2R. Minor optimization that I have made as I was digging through this.